CodeGen/Generators/QuantityJsonFilesParser.cs

```diff
@@ -109,6 +111,72 @@ private static void AddPrefixUnits(Quantity quantity)
             quantity.Units = quantity.Units.Concat(unitsToAdd).ToArray();
         }
 
+        private static BaseUnits? GetBaseUnitFromPrefix(Unit unit, Prefix prefix)
+        {
+            if (unit.BaseUnits is null)
+                return null;
+            BaseUnits? dup = JsonConvert.DeserializeObject<BaseUnits>(JsonConvert.SerializeObject(unit.BaseUnits));
+            string[] words = ToWords(unit.SingularName);
+            //First Word is not Enough (Ex. PoundMole)
+            for (int i = 0; i < words.Length; i++)
+            {
+                BaseUnits? ret = AddPrefixToBaseUnits(dup, string.Join("", words.Take(i + 1)), prefix);
+                if (ret is not null)
+                    return ret;
+            }
+            
+            return null;
+        }
+
+        private static BaseUnits? AddPrefixToBaseUnits(BaseUnits? ret, string firstWord, Prefix prefix)
+        {
+            if (ret is null)
+                return null;
+            else if (ret.N is not null && ret.N == firstWord)
+            {
+                ret.N = prefix.ToString() + ret.N.ToCamelCase();
+                return ret;
+            }
+            else if (ret.I is not null && ret.I == firstWord)
+            {
+                ret.I = prefix.ToString() + ret.I.ToCamelCase();
+                return ret;
+            }
+            else if (ret.L is not null && ret.L == firstWord)
+            {
+                ret.L = prefix.ToString() + ret.L.ToCamelCase();
+                return ret;
+            }
+            else if (ret.J is not null && ret.J == firstWord)
+            {
+                ret.J = prefix.ToString() + ret.J.ToCamelCase();
+                return ret;
+            }
+            else if (ret.M is not null && ret.M == firstWord)
+            {
+                ret.M = prefix.ToString() + ret.M.ToCamelCase();
+                return ret;
+            }
+            else if (ret.Θ is not null && ret.Θ == firstWord)
+            {
+                ret.Θ = prefix.ToString() + ret.Θ.ToCamelCase();
+                return ret;
+            }
+            else if (ret.T is not null && ret.T == firstWord)
+            {
+                ret.T = prefix.ToString() + ret.T.ToCamelCase();
+                return ret;
+            }
+            else
+                return null;
+        }
+
+        public static string[] ToWords(string str)
+        {
+            return Regex.Replace(str, "[a-z][A-Z]", m => $"{m.Value[0]} {m.Value[1]}")
+                .Split(' ');
+        }
+
```

Comment:
It is an interesting approach, but it only solves 82 of 216 prefix units.
I pushed a commit with some debug console output.

We might be able improve the name matching, but there is risk of matching incorrectly and I don't think we can get to 100% coverage with this approach for units like Farad and StandardGravity anyway.


```
Assigned base units to 134 prefix units, failed to assign for 82 prefix units:
---
StandardGravity           L=Meter T=Second                            Prefixes: [Milli]
Farad                     I=Ampere L=Meter M=Kilogram T=Second        Prefixes: [Pico, Nano, Micro, Milli, Kilo, Mega]
Volt                      I=Ampere L=Meter M=Kilogram T=Second        Prefixes: [Nano, Micro, Milli, Kilo, Mega]
VoltPerSecond             I=Ampere L=Meter M=Kilogram T=Second        Prefixes: [Micro, Milli, Kilo, Mega]
VoltPerMicrosecond        I=Ampere L=Meter M=Kilogram T=Microsecond   Prefixes: [Micro, Milli, Kilo, Mega]
VoltPerMinute             I=Ampere L=Meter M=Kilogram T=Minute        Prefixes: [Micro, Milli, Kilo, Mega]
VoltPerHour               I=Ampere L=Meter M=Kilogram T=Hour          Prefixes: [Micro, Milli, Kilo, Mega]
Joule                     L=Meter M=Kilogram T=Second                 Prefixes: [Milli, Kilo, Mega, Giga, Tera, Peta]
JoulePerCubicMeter        L=Meter M=Kilogram T=Second                 Prefixes: [Kilo, Mega, Giga, Tera, Peta]
Newton                    L=Meter M=Kilogram T=Second                 Prefixes: [Micro, Milli, Deca, Kilo, Mega]
WattPerSquareMeter        M=Kilogram T=Second                         Prefixes: [Nano, Micro, Milli, Centi, Deci, Kilo]
StandardGravitiesPerSecond L=Meter T=Second                           Prefixes: [Milli]
Watt                      L=Meter M=Kilogram T=Second                 Prefixes: [Femto, Pico, Nano, Micro, Milli, Deci, Deca, Kilo, Mega, Giga, Tera, Peta]
WattPerCubicMeter         L=Meter M=Kilogram T=Second                 Prefixes: [Pico, Nano, Micro, Milli, Deci, Deca, Kilo, Mega, Giga, Tera]
Pascal                    L=Meter M=Kilogram T=Second                 Prefixes: [Micro, Milli, Deca, Hecto, Kilo, Mega, Giga]
CubicMeter                L=Meter                                     Prefixes: [Hecto, Kilo]
---
```


I guess it is better than nothing though? We could go with this.

One obvious solution is to extend the JSON to explicitly map base units for each individual prefix unit in the JSON, but I think it will be messy and more cumbersome to add new units for contributors. I don't see enough utility value or upside to justify this effort.

Reply:
I couldn't push changes to your master branch, so I created a pull request to update your pull request: https://github.com/bdbonazz/UnitsNet/pull/1

Reply:
@tmilnthorp what do you think of this approach? Do you see any better ways to achieve this?

Reply:
> I don't think we can get to 100% coverage

I don't think that too, but I cannot see at the moment any case of possible incorrect match

> One obvious solution is to extend the JSON to explicitly map base units for each individual prefix unit in the JSON, but I think it will be messy and more cumbersome to add new units for contributors. I don't see enough utility value or upside to justify this effort.

I totally agree with that. I'm either non totally sure this is possible anyway
Ex. MicroPascal could etiher be "g/m*s^2" or "kg/km*s^2"

Reply:
@tmilnthorp Any input on this? I'm inclined to merge it, it seems like a solid improvement although not 100%. Can't spot any mistakes in the automatically determined base units.

Reply:
@tmilnthorp Just a nudge on your opinion here before I merge.

Reply:
> @tmilnthorp Just a nudge on your opinion here before I merge.

Sorry. Work and personal life got busy. Let me take some time to look in the next day or two. If you don’t hear by then go for it.

Reply:
@tmilnthorp 
I hope I will be clear enough, English is not my first language
I don't know why I cannot reply to your other comment
"So just thinking about this - wouldn't the base units just be the same as the ones without the prefixes? Per your example:"
No, this is exactly what this pull request is about
I'm taking MetersPerSecondSquared [L=Meter T=Second] and saying that the prefix Nano "NanoMetersPerSecondSquared" changes that to [L=NanoMeter T=Second]
So I was saying I don't know how (or even if it's possible) to uniquely link MicroPascal to [M=Kilogram L=KiloMeter T=Second] or [M=Gram L=Meter T=Second]
And Pascal is even an easy example, Farad could be even more impossible in my opinion
Farad in SI is (s^4*A^2)/(m^2*kg). The only "correct" way to write MilliFarad is (s^4*A^2)/(m^2*g) but more often that not this is wrong. Often (s^4*A^2)/(m^2*kg) are the correct units and people use MilliFarad cause the results are small